lib/hooks.js

```diff
@@ -96,11 +96,16 @@ async function init (options) {
       // for the `request` hook, merge and return options data
       if (hookName === 'request') {
         // merge results of returned hook data
-        const finalData = Object.assign({}, data.options, {
-          queryParams: { ...data.options.queryParams },
-          pathParams: { ...data.options.pathParams },
-          headers: { ...data.options.headers }
-        })
+        const finalData = { ...data.options }
+        if (data.options.queryParams) {
+          finalData.queryParams = { ...data.options.queryParams }
+        }
+        if (data.options.pathParams) {
+          finalData.pathParams = { ...data.options.pathParams }
+        }
+        if (data.options.headers) {
+          finalData.headers = { ...data.options.headers }
+        }
```

Comment:
This means no `finalData.queryParams` object will be created if none were passed to the request. Previously it would create an empty object if none were passed to the request.